PSKoans/Koans/Constructs and Patterns/AboutClasses.Koans.ps1

```diff
@@ -0,0 +1,985 @@
+                        if ($object -isnot [VideoGame]) {
+                            return $false
+                        }
+
+                        # Objects are considered equal if the expression below returns true.
+                        return $this.Name -eq $object.Name -and $this.Publisher -eq $object.Publisher
+                    }
+                }
+
+                $game1 = [VideoGame]@{
+                    Name      = 'Grim Fandango'
+                    Publisher = 'LucasArts'
+                    Rating    = 94
+                }
+                $game2 = [VideoGame]@{
+                    Name      = 'Day of the Tentacle'
+                    Publisher = 'Double Fine Productions'
+                    Rating    = 86
+                }
+
+                $game1 -eq $game2
+            }
+
+            $____ | Should -Be (Start-Job -ScriptBlock $script | Receive-Job -Wait)
+        }
+
+        It 'can support both equality and comparison' {
+            $script = {
+                class VideoGame : IComparable, IEquatable[Object] {
+                    [string] $Name
+                    [string] $Publisher
+                    [int]    $Rating
+
+                    [int] CompareTo([Object] $object) {
+                        if ($this.Rating -eq $object.Rating) {
+                            return 0
+                        } elseif ($this.Rating -gt $object.Rating) {
+                            return 1
+                        } else {
+                            return -1
+                        }
+                    }
+
+                    [bool] Equals([Object] $object) {
+                        if ($object -isnot [VideoGame]) {
+                            return $false
+                        }
+                        <#
+                            If equality is based on the same comparison as IComparable, that is CompareTo being 0,
+                            the following expression might be used:
+
+                                return $this.CompareTo($object) -eq 0
+                        #>
+                        return $this.Name -eq $object.Name -and $this.Publisher -eq $object.Publisher
+                    }
+                }
+
+                $game1 = [VideoGame]@{
+                    Name      = 'Grim Fandango'
+                    Publisher = 'LucasArts'
+                    Rating    = 94
+                }
+                $game2 = [VideoGame]@{
+                    Name      = 'Day of the Tentacle'
+                    Publisher = 'Double Fine Productions'
+                    Rating    = 86
+                }
+
+                $game1 -eq $game2
+            }
+
+            $____ | Should -Be (Start-Job -ScriptBlock $script | Receive-Job -Wait)
+        }
+    }
+
+    Context 'Casting' {
+        <#
+            Casting a value to an instance of a PowerShell class might be supported by implementing a constructor for
+            that type.
+        #>
```

Comment:
This wording needs some work... Perhaps something more like..?
```
Casting a value to a PowerShell class type can be supported by adding a constructor which takes an
object of the value's type as an argument.
```

Reply:
Also, I see you covering `op_Implicit` below; if I recall correctly you can do a reverse implementation to convert _from_ another type to the PowerShell class type?

Might be good to add that example as well.